libs/fleet-mcp/src/fleet_mcp/services/agent_service.py

```diff
@@ -64,16 +80,24 @@ async def list_agents(
             project_filter_lower = project_filter.lower()
             agents = [a for a in agents if a.project.lower() == project_filter_lower]
 
+        # Always collect metadata if metadata_repo is available
+        if self.metadata_repo is not None:
+            for agent in agents:
+                metadata = await self.metadata_repo.collect_metadata(agent.workspace_id)
+                # Convert WorkspaceMetadata to dict for Agent model
+                agent.metadata = metadata.model_dump()
+
         return agents
 
-    async def get_agent(self, name: str) -> Agent:
+    async def get_agent(self, name: str, include_metadata: bool = True) -> Agent:
```

Comment:
List agent will always include metadata, this is not optional or configurable. Remove any branching logic that does otherwise